WordPress/gutenberg
# Try/publish in site editor

## What?
Adds ability to set status, publish date and password in site editor.

<img width="1015" alt="image" src="https://github.com/WordPress/gutenberg/assets/1072756/679fb398-ae8f-4297-89f9-ef03eae501c0">

<img width="989" alt="image" src="https://github.com/WordPress/gutenberg/assets/1072756/6a73236e-9b04-4819-b463-ebb65ec0a0c6">


## Why?
Resolves https://github.com/WordPress/gutenberg/issues/49873

## How?
WIP. 

## Testing Instructions
1. Open site editor
2. Go to `pages` and select a page to edit
3. Edit the page and open up the page inspector and notice the new summary panel
4. Try changing the status from draft to published
5. Try updating the publish date to a point in the future and notice status changes to scheduled
6. Try updating the publish date to a point in the past and notice status changes back to published
7. Hit save and check to see if status change is persisted

### Testing Instructions for Keyboard


## Screenshots or screencast
